packages/babel-parser/src/parser/comments.js

```diff
@@ -50,10 +67,10 @@ function setTrailingComments(node: Node, comments: Array<Comment>) {
  * @param {Node} node
  * @param {Array<Comment>} comments
  */
-export function setInnerComments(node: Node, comments: Array<Comment> | void) {
+export function setInnerComments(node: Node, comments: Array<Comment>) {
   if (node.innerComments === undefined) {
     node.innerComments = comments;
-  } else if (comments !== undefined) {
```

Comment:
This is just to align the signature to the other two functions: since they are really similar it's easier to keep the implementation the same.